src/js/background/assignManager.js

```diff
@@ -233,9 +238,11 @@ window.assignManager = {
         return {};
       }
     }
+    const replaceTabEnabled = await this.storageArea.getReplaceTabEnabled();
     const removeTab = backgroundLogic.NEW_TAB_PAGES.has(tab.url)
       || (messageHandler.lastCreatedTab
-        && messageHandler.lastCreatedTab.id === tab.id);
+        && messageHandler.lastCreatedTab.id === tab.id)
+      || replaceTabEnabled;
```

Comment:
Essentially this one line is the only functionality change. The other code of this PR is there solely to add the option to turn this functionality on or off. The removal of the current tab was already implemented but was only used for blank tabs.